apps/dashboard/app/apps/user_preferences.rb

```diff
@@ -0,0 +1,48 @@
+class UserPreferences
+    require 'yaml'
+  
+    def initialize()
+        FileUtils.mkdir_p("~/.config/ondemand/") unless File.exists?("~/.config/ondemand/")
```

Comment:
I think this `mkdir_p` belongs in the `preferences_file` method so that we don't have this same file location in two places.